src/dbnode/client/connection_pool.go

```diff
@@ -181,19 +196,28 @@ func (p *connPool) connectEvery(interval time.Duration, stutter time.Duration) {
 				defer wg.Done()
 
 				// Create connection
-				channel, client, err := p.newConn(channelName, address, p.opts)
+				channel, thriftClient, err := p.newConn(channelName, address, p.opts)
 				if err != nil {
 					log.Debug("could not connect", zap.String("host", address), zap.Error(err))
 					return
 				}
 
+				client := p.newConnClient(thriftClient)
```

Comment:
nit: Should we do this after checking if it supports snappy?

Reply:
Unfortunately we need it for the first check (i.e. the client needs to do an uncompressed request to see if the server supports compression or not).

Reply:
What if this information is stored as a property in the placement?